src/commands/execution_ctx.c

```diff
@@ -20,21 +21,26 @@ static ExecutionType _GetExecutionTypeFromAST(AST *ast) {
 static ExecutionCtx *_ExecutionCtx_New(AST *ast, ExecutionPlan *plan,
   		ExecutionType exec_type) {
 	ExecutionCtx *exec_ctx = rm_malloc(sizeof(ExecutionCtx));
-	exec_ctx->ast = ast;
-	exec_ctx->plan = plan;
+
+	exec_ctx->ast       = ast;
+	exec_ctx->plan      = plan;
+	exec_ctx->cached    = false;
 	exec_ctx->exec_type = exec_type;
-	exec_ctx->cached = false;
+
 	return exec_ctx;
 }
 
 ExecutionCtx *ExecutionCtx_Clone(ExecutionCtx *orig) {
 	ExecutionCtx *execution_ctx = rm_malloc(sizeof(ExecutionCtx));
-	execution_ctx->ast = AST_ShallowCopy(orig->ast);
-	// Set the AST copy in thread local storage.
-	QueryCtx_SetAST(execution_ctx->ast);
-	execution_ctx->plan = ExecutionPlan_Clone(orig->plan);
+
+	execution_ctx->ast       = AST_ShallowCopy(orig->ast);
+	execution_ctx->plan      = ExecutionPlan_Clone(orig->plan);
+	execution_ctx->cached    = orig->cached;
 	execution_ctx->exec_type = orig->exec_type;
-	execution_ctx->cached = orig->cached;
+
+	// set the AST copy in thread local storage
+	QueryCtx_SetAST(execution_ctx->ast);
```

Comment:
An assertion depends on this being called before `ExecutionPlan_Clone` (line 37).